common/src/BasePlugin.php

```diff
@@ -218,4 +242,85 @@ private function isUniqueAttribute(array $array, string $value): bool {
     private function isValidGtin(string $value): bool {
         return preg_match('/^\d{8}(?:\d{4,6})?$/', $value);
     }
+
+    public function showCustomNotice() {
+        $class = 'notice notice-info is-dismissible ' . $this->getCustomNoticeClass();
+        $message = $this->getCustomNoticeText();
+        if (!empty($message)) {
+            printf('<div class="%s"><p>%s</p></div>', esc_attr($class), $message);
+        }
+    }
+
+    public function getCustomNoticeText() {
+        $readme = wp_remote_fopen($this->getPluginReadme());
```

Comment:
Why are you downloading the readme from the Subversion repository? It will be included in the local plugin directory. Just run `make` to generate it.